From 372ae46c7e7c1281eb40625f2702eafab7c1f2a9 Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@notvalid.com>
Date: Wed, 14 Oct 2026 00:27:10 -0700
Subject: [PATCH] 003-remove-ecash-activations-and-enforce-from-genesis

Enforce the selected inherited consensus rules from genesis for the development chain and remove obsolete eCash activation switches. This is not a final V2 network profile.

Original source commit: d1e3c8ea0e2125d987c64d812559b7f41e4856a6
---
 src/consensus/activation.cpp        | 120 -------------------
 src/consensus/activation.h          |  39 -------
 src/consensus/params.h              |  23 ----
 src/consensus/tx_verify.cpp         |  22 ++--
 src/init.cpp                        |   2 -
 src/kernel/chainparams.cpp          | 138 +++-------------------
 src/minerfund.cpp                   |   6 +-
 src/node/miner.cpp                  |   6 +-
 src/policy/block/minerfund.cpp      |   4 +-
 src/policy/block/stakingrewards.cpp |   5 +-
 src/pow/aserti32d.cpp               |   2 +-
 src/pow/pow.cpp                     |  37 +++---
 src/pow/test/aserti32d_tests.cpp    | 171 ----------------------------
 src/rpc/mining.cpp                  |  11 +-
 src/test/CMakeLists.txt             |   2 -
 src/test/activation_tests.cpp       |  49 +-------
 src/test/mempool_tests.cpp          |   6 +-
 src/test/minerfund_tests.cpp        |  34 +-----
 src/test/policy_block_tests.cpp     |  52 +--------
 src/test/transaction_tests.cpp      |   7 +-
 src/validation.cpp                  | 130 +++++----------------
 21 files changed, 89 insertions(+), 777 deletions(-)

diff --git a/src/consensus/activation.cpp b/src/consensus/activation.cpp
index 923aafaea..b3f8028cb 100644
--- a/src/consensus/activation.cpp
+++ b/src/consensus/activation.cpp
@@ -7,123 +7,3 @@
 #include <chain.h>
 #include <common/args.h>
 #include <consensus/params.h>
-
-bool IsUAHFenabled(const Consensus::Params &params, int nHeight) {
-    return nHeight >= params.uahfHeight;
-}
-
-bool IsUAHFenabled(const Consensus::Params &params,
-                   const CBlockIndex *pindexPrev) {
-    if (pindexPrev == nullptr) {
-        return false;
-    }
-
-    return IsUAHFenabled(params, pindexPrev->nHeight);
-}
-
-bool IsDAAEnabled(const Consensus::Params &params, int nHeight) {
-    return nHeight >= params.daaHeight;
-}
-
-bool IsDAAEnabled(const Consensus::Params &params,
-                  const CBlockIndex *pindexPrev) {
-    if (pindexPrev == nullptr) {
-        return false;
-    }
-
-    return IsDAAEnabled(params, pindexPrev->nHeight);
-}
-
-bool IsMagneticAnomalyEnabled(const Consensus::Params &params,
-                              int32_t nHeight) {
-    return nHeight >= params.magneticAnomalyHeight;
-}
-
-bool IsMagneticAnomalyEnabled(const Consensus::Params &params,
-                              const CBlockIndex *pindexPrev) {
-    if (pindexPrev == nullptr) {
-        return false;
-    }
-
-    return IsMagneticAnomalyEnabled(params, pindexPrev->nHeight);
-}
-
-static bool IsGravitonEnabled(const Consensus::Params &params,
-                              int32_t nHeight) {
-    return nHeight >= params.gravitonHeight;
-}
-
-bool IsGravitonEnabled(const Consensus::Params &params,
-                       const CBlockIndex *pindexPrev) {
-    if (pindexPrev == nullptr) {
-        return false;
-    }
-
-    return IsGravitonEnabled(params, pindexPrev->nHeight);
-}
-
-static bool IsPhononEnabled(const Consensus::Params &params, int32_t nHeight) {
-    return nHeight >= params.phononHeight;
-}
-
-bool IsPhononEnabled(const Consensus::Params &params,
-                     const CBlockIndex *pindexPrev) {
-    if (pindexPrev == nullptr) {
-        return false;
-    }
-
-    return IsPhononEnabled(params, pindexPrev->nHeight);
-}
-
-static bool IsAxionEnabled(const Consensus::Params &params, int32_t nHeight) {
-    return nHeight >= params.axionHeight;
-}
-
-bool IsAxionEnabled(const Consensus::Params &params,
-                    const CBlockIndex *pindexPrev) {
-    if (pindexPrev == nullptr) {
-        return false;
-    }
-
-    return IsAxionEnabled(params, pindexPrev->nHeight);
-}
-
-bool IsWellingtonEnabled(const Consensus::Params &params, int32_t nHeight) {
-    return nHeight >= params.wellingtonHeight;
-}
-
-bool IsWellingtonEnabled(const Consensus::Params &params,
-                         const CBlockIndex *pindexPrev) {
-    if (pindexPrev == nullptr) {
-        return false;
-    }
-
-    return IsWellingtonEnabled(params, pindexPrev->nHeight);
-}
-
-bool IsCowperthwaiteEnabled(const Consensus::Params &params, int32_t nHeight) {
-    return nHeight >= params.cowperthwaiteHeight;
-}
-
-bool IsCowperthwaiteEnabled(const Consensus::Params &params,
-                            const CBlockIndex *pindexPrev) {
-    if (pindexPrev == nullptr) {
-        return false;
-    }
-
-    return IsCowperthwaiteEnabled(params, pindexPrev->nHeight);
-}
-
-bool IsMengerEnabled(const Consensus::Params &params, int64_t nMedianTimePast) {
-    return nMedianTimePast >= gArgs.GetIntArg("-mengeractivationtime",
-                                              params.mengerActivationTime);
-}
-
-bool IsMengerEnabled(const Consensus::Params &params,
-                     const CBlockIndex *pindexPrev) {
-    if (pindexPrev == nullptr) {
-        return false;
-    }
-
-    return IsMengerEnabled(params, pindexPrev->GetMedianTimePast());
-}
diff --git a/src/consensus/activation.h b/src/consensus/activation.h
index 2c9f724d0..bfe50549f 100644
--- a/src/consensus/activation.h
+++ b/src/consensus/activation.h
@@ -13,43 +13,4 @@ namespace Consensus {
 struct Params;
 }
 
-/** Check if UAHF has activated. */
-bool IsUAHFenabled(const Consensus::Params &params,
-                   const CBlockIndex *pindexPrev);
-bool IsUAHFenabled(const Consensus::Params &params, int nHeight);
-
-/** Check if DAA HF has activated. */
-bool IsDAAEnabled(const Consensus::Params &params,
-                  const CBlockIndex *pindexPrev);
-bool IsDAAEnabled(const Consensus::Params &params, int nHeight);
-
-/** Check if Nov 15, 2018 HF has activated using block height. */
-bool IsMagneticAnomalyEnabled(const Consensus::Params &params, int32_t nHeight);
-/** Check if Nov 15, 2018 HF has activated using previous block index. */
-bool IsMagneticAnomalyEnabled(const Consensus::Params &params,
-                              const CBlockIndex *pindexPrev);
-
-/** Check if Nov 15th, 2019 protocol upgrade has activated. */
-bool IsGravitonEnabled(const Consensus::Params &params,
-                       const CBlockIndex *pindexPrev);
-
-/** Check if May 15th, 2020 protocol upgrade has activated. */
-bool IsPhononEnabled(const Consensus::Params &params,
-                     const CBlockIndex *pindexPrev);
-
-/** Check if November 15th, 2020 protocol upgrade has activated. */
-bool IsAxionEnabled(const Consensus::Params &params,
-                    const CBlockIndex *pindexPrev);
-
-/** Check if May 15th, 2023 protocol upgrade has activated. */
-bool IsWellingtonEnabled(const Consensus::Params &params, int32_t nHeight);
-/** Check if May 15th, 2023 protocol upgrade has activated. */
-bool IsWellingtonEnabled(const Consensus::Params &params,
-                         const CBlockIndex *pindexPrev);
-
-/** Check if Nov 15th, 2023 protocol upgrade has activated. */
-bool IsCowperthwaiteEnabled(const Consensus::Params &params, int32_t nHeight);
-bool IsCowperthwaiteEnabled(const Consensus::Params &params,
-                            const CBlockIndex *pindexPrev);
-
 #endif // BITCOIN_CONSENSUS_ACTIVATION_H
diff --git a/src/consensus/params.h b/src/consensus/params.h
index 1829cbf6a..eeef81183 100644
--- a/src/consensus/params.h
+++ b/src/consensus/params.h
@@ -45,29 +45,6 @@ struct Params {
     int BIP66Height;
     /** Block height at which CSV (BIP68, BIP112 and BIP113) becomes active */
     int CSVHeight;
-    /** Block height at which UAHF kicks in */
-    int uahfHeight;
-    /** Block height at which the new DAA becomes active */
-    int daaHeight;
-    /** Block height at which the magnetic anomaly activation becomes active */
-    int magneticAnomalyHeight;
-    /** Block height at which the graviton activation becomes active */
-    int gravitonHeight;
-    /** Block height at which the phonon activation becomes active */
-    int phononHeight;
-    /** Block height at which the axion activation becomes active */
-    int axionHeight;
-    /** Block height at which the wellington activation becomes active */
-    int wellingtonHeight;
-    /** Block height at which the Cowperthwaite activation becomes active */
-    int cowperthwaiteHeight;
-    /**
-     * Unix time used for MTP activation of 15 November 2026 12:00:00 UTC
-     * upgrade
-     */
-    int mengerActivationTime;
-    /** Unix time used for MTP activation of 15 May 2027 12:00:00 UTC upgrade */
-    int sandersActivationTime;
 
     /** Enable or disable the miner fund by default */
     bool enableMinerFund;
diff --git a/src/consensus/tx_verify.cpp b/src/consensus/tx_verify.cpp
index d680fba3e..5f9bf7450 100644
--- a/src/consensus/tx_verify.cpp
+++ b/src/consensus/tx_verify.cpp
@@ -49,21 +49,17 @@ bool ContextualCheckTransaction(const Consensus::Params &params,
                              "bad-txns-nonfinal", "non-final transaction");
     }
 
-    if (IsMagneticAnomalyEnabled(params, nHeight)) {
-        // Size limit
-        if (::GetSerializeSize(tx) < MIN_TX_SIZE) {
-            return state.Invalid(TxValidationResult::TX_CONSENSUS,
-                                 "bad-txns-undersize");
-        }
+    // Size limit
+    if (::GetSerializeSize(tx) < MIN_TX_SIZE) {
+        return state.Invalid(TxValidationResult::TX_CONSENSUS,
+                             "bad-txns-undersize");
     }
 
-    if (IsWellingtonEnabled(params, nHeight)) {
-        // Restrict version to 1 and 2
-        if (tx.nVersion > CTransaction::MAX_VERSION ||
-            tx.nVersion < CTransaction::MIN_VERSION) {
-            return state.Invalid(TxValidationResult::TX_CONSENSUS,
-                                 "bad-txns-version");
-        }
+    // Restrict version to 1 and 2
+    if (tx.nVersion > CTransaction::MAX_VERSION ||
+        tx.nVersion < CTransaction::MIN_VERSION) {
+        return state.Invalid(TxValidationResult::TX_CONSENSUS,
+                             "bad-txns-version");
     }
 
     return true;
diff --git a/src/init.cpp b/src/init.cpp
index f817aed2f..3aa057fe6 100644
--- a/src/init.cpp
+++ b/src/init.cpp
@@ -512,8 +512,6 @@ void SetupServerArgs(NodeContext &node) {
         "-rootcertificates=<file>",
         "-splash",
         "-uiplatform",
-        // TODO remove after the November 2026 upgrade
-        "-mengeractivationtime",
     };
 
     // Set all of the args and their help
diff --git a/src/kernel/chainparams.cpp b/src/kernel/chainparams.cpp
index 7be70c725..b3046f98e 100644
--- a/src/kernel/chainparams.cpp
+++ b/src/kernel/chainparams.cpp
@@ -84,18 +84,12 @@ public:
     explicit CMainParams(const ChainOptions &opts) {
         m_chain_type = ChainType::MAIN;
         consensus.nSubsidyHalvingInterval = 210000;
-        // 00000000000000ce80a7e057163a4db1d5ad7b20fb6f598c9597b9665c8fb0d4 -
-        // April 1, 2012
-        consensus.BIP16Height = 173805;
-        consensus.BIP34Height = 227931;
-        consensus.BIP34Hash = *BlockHash::FromHex(
-            "000000000000024b89b42a942fe0d9fea3bb44ab7bd1b19115dd6a759c0808b8");
-        // 000000000000000004c2b624ed5d7756c508d90fd0da2c7c679febfa6c4735f0
-        consensus.BIP65Height = 388381;
-        // 00000000000000000379eaa19dce8c9b722d46ae6a57c2f1a988119488b50931
-        consensus.BIP66Height = 363725;
-        // 000000000000000004a1b34462cb8aeebd5799177f7a29cf28f2d1961716b5b5
-        consensus.CSVHeight = 419328;
+        consensus.BIP16Height = 0;
+        consensus.BIP34Height = 0;
+        consensus.BIP34Hash = BlockHash();
+        consensus.BIP65Height = 0;
+        consensus.BIP66Height = 0;
+        consensus.CSVHeight = 0;
         consensus.powLimit = uint256S(
             "00000000ffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
         // two weeks
@@ -122,36 +116,6 @@ public:
         consensus.defaultAssumeValid =
             ChainParamsConstants::MAINNET_DEFAULT_ASSUME_VALID;
 
-        // August 1, 2017 hard fork
-        consensus.uahfHeight = 478558;
-
-        // November 13, 2017 hard fork
-        consensus.daaHeight = 504031;
-
-        // November 15, 2018 hard fork
-        consensus.magneticAnomalyHeight = 556766;
-
-        // November 15, 2019 protocol upgrade
-        consensus.gravitonHeight = 609135;
-
-        // May 15, 2020 12:00:00 UTC protocol upgrade
-        consensus.phononHeight = 635258;
-
-        // Nov 15, 2020 12:00:00 UTC protocol upgrade
-        consensus.axionHeight = 661647;
-
-        // May 15, 2023 12:00:00 UTC protocol upgrade
-        consensus.wellingtonHeight = 792116;
-
-        // Nov 15, 2023 12:00:00 UTC protocol upgrade
-        consensus.cowperthwaiteHeight = 818669;
-
-        // Nov 15, 2026 12:00:00 UTC protocol upgrade
-        consensus.mengerActivationTime = 1794744000;
-
-        // May 15, 2027 12:00:00 UTC protocol upgrade
-        consensus.sandersActivationTime = 1810382400;
-
         /**
          * The message start string is designed to be unlikely to occur in
          * normal data. The characters are rarely used upper ASCII, not valid as
@@ -256,17 +220,12 @@ public:
     explicit CTestNetParams(const ChainOptions &opts) {
         m_chain_type = ChainType::TESTNET;
         consensus.nSubsidyHalvingInterval = 210000;
-        // 00000000040b4e986385315e14bee30ad876d8b47f748025b26683116d21aa65
-        consensus.BIP16Height = 514;
-        consensus.BIP34Height = 21111;
-        consensus.BIP34Hash = *BlockHash::FromHex(
-            "0000000023b3a96d3484e5abb3755c413e7d41500f8e2a5c3f0dd01299cd8ef8");
-        // 00000000007f6655f22f98e72ed80d8b06dc761d5da09df0fa1dc4be4f861eb6
-        consensus.BIP65Height = 581885;
-        // 000000002104c8c45e99a8853285a3b592602a3ccde2b832481da85e9e4ba182
-        consensus.BIP66Height = 330776;
-        // 00000000025e930139bac5c6c31a403776da130831ab85be56578f3fa75369bb
-        consensus.CSVHeight = 770112;
+        consensus.BIP16Height = 0;
+        consensus.BIP34Height = 0;
+        consensus.BIP34Hash = BlockHash();
+        consensus.BIP65Height = 0;
+        consensus.BIP66Height = 0;
+        consensus.CSVHeight = 0;
         consensus.powLimit = uint256S(
             "00000000ffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
         // two weeks
@@ -293,36 +252,6 @@ public:
         consensus.defaultAssumeValid =
             ChainParamsConstants::TESTNET_DEFAULT_ASSUME_VALID;
 
-        // August 1, 2017 hard fork
-        consensus.uahfHeight = 1155875;
-
-        // November 13, 2017 hard fork
-        consensus.daaHeight = 1188697;
-
-        // November 15, 2018 hard fork
-        consensus.magneticAnomalyHeight = 1267996;
-
-        // November 15, 2019 protocol upgrade
-        consensus.gravitonHeight = 1341711;
-
-        // May 15, 2020 12:00:00 UTC protocol upgrade
-        consensus.phononHeight = 1378460;
-
-        // Nov 15, 2020 12:00:00 UTC protocol upgrade
-        consensus.axionHeight = 1421481;
-
-        // May 15, 2023 12:00:00 UTC protocol upgrade
-        consensus.wellingtonHeight = 1556117;
-
-        // Nov 15, 2023 12:00:00 UTC protocol upgrade
-        consensus.cowperthwaiteHeight = 1584485;
-
-        // Nov 15, 2026 12:00:00 UTC protocol upgrade
-        consensus.mengerActivationTime = 1794744000;
-
-        // May 15, 2027 12:00:00 UTC protocol upgrade
-        consensus.sandersActivationTime = 1810382400;
-
         diskMagic[0] = 0x0b;
         diskMagic[1] = 0x11;
         diskMagic[2] = 0x09;
@@ -411,17 +340,12 @@ public:
     explicit CRegTestParams(const ChainOptions &opts) {
         m_chain_type = ChainType::REGTEST;
         consensus.nSubsidyHalvingInterval = 150;
-        // always enforce P2SH BIP16 on regtest
         consensus.BIP16Height = 0;
-        // BIP34 activated on regtest (Used in functional tests)
-        consensus.BIP34Height = 500;
+        consensus.BIP34Height = 0;
         consensus.BIP34Hash = BlockHash();
-        // BIP65 activated on regtest (Used in functional tests)
-        consensus.BIP65Height = 1351;
-        // BIP66 activated on regtest (Used in functional tests)
-        consensus.BIP66Height = 1251;
-        // CSV activated on regtest (Used in functional tests)
-        consensus.CSVHeight = 576;
+        consensus.BIP65Height = 0;
+        consensus.BIP66Height = 0;
+        consensus.CSVHeight = 0;
         consensus.powLimit = uint256S(
             "7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff");
         // two weeks
@@ -446,36 +370,6 @@ public:
         // valid.
         consensus.defaultAssumeValid = BlockHash();
 
-        // UAHF is always enabled on regtest.
-        consensus.uahfHeight = 0;
-
-        // November 13, 2017 hard fork is always on on regtest.
-        consensus.daaHeight = 0;
-
-        // November 15, 2018 hard fork is always on on regtest.
-        consensus.magneticAnomalyHeight = 0;
-
-        // November 15, 2019 protocol upgrade
-        consensus.gravitonHeight = 0;
-
-        // May 15, 2020 12:00:00 UTC protocol upgrade
-        consensus.phononHeight = 0;
-
-        // Nov 15, 2020 12:00:00 UTC protocol upgrade
-        consensus.axionHeight = 0;
-
-        // May 15, 2023 12:00:00 UTC protocol upgrade
-        consensus.wellingtonHeight = 0;
-
-        // Nov 15, 2023 12:00:00 UTC protocol upgrade
-        consensus.cowperthwaiteHeight = 0;
-
-        // Nov 15, 2026 12:00:00 UTC protocol upgrade
-        consensus.mengerActivationTime = 1794744000;
-
-        // May 15, 2027 12:00:00 UTC protocol upgrade
-        consensus.sandersActivationTime = 1810382400;
-
         diskMagic[0] = 0xfa;
         diskMagic[1] = 0xbf;
         diskMagic[2] = 0xb5;
diff --git a/src/minerfund.cpp b/src/minerfund.cpp
index d69f32a8f..01d1f5248 100644
--- a/src/minerfund.cpp
+++ b/src/minerfund.cpp
@@ -16,16 +16,12 @@
 /**
  * Percentage of the block reward to be sent to the fund.
  */
-static constexpr int LEGACY_MINER_FUND_RATIO = 8;
 static constexpr int MINER_FUND_RATIO = 32;
 
 Amount GetMinerFundAmount(const Consensus::Params &params,
                           const Amount &coinbaseValue,
                           const CBlockIndex *pprev) {
-    const int minerFundRatio = IsCowperthwaiteEnabled(params, pprev)
-                                   ? MINER_FUND_RATIO
-                                   : LEGACY_MINER_FUND_RATIO;
-    return minerFundRatio * coinbaseValue / 100;
+    return MINER_FUND_RATIO * coinbaseValue / 100;
 }
 
 static CTxDestination BuildDestination(const std::string &dest) {
diff --git a/src/node/miner.cpp b/src/node/miner.cpp
index 8f7d3dc3d..2ae0be81e 100644
--- a/src/node/miner.cpp
+++ b/src/node/miner.cpp
@@ -142,10 +142,8 @@ BlockAssembler::CreateNewBlock(const CScript &scriptPubKeyIn) {
     }
 
     // Canonical ordering is naturally enforced when using the radix tree
-    if (!shouldAddFinalizedTxs &&
-        IsMagneticAnomalyEnabled(consensusParams, pindexPrev)) {
-        // If magnetic anomaly is enabled, we make sure transaction are
-        // canonically ordered.
+    if (!shouldAddFinalizedTxs) {
+        // We make sure transactions are canonically ordered.
         std::sort(std::begin(pblocktemplate->entries) + 1,
                   std::end(pblocktemplate->entries),
                   [](const CBlockTemplateEntry &a, const CBlockTemplateEntry &b)
diff --git a/src/policy/block/minerfund.cpp b/src/policy/block/minerfund.cpp
index db1f452b4..0ddfa1b92 100644
--- a/src/policy/block/minerfund.cpp
+++ b/src/policy/block/minerfund.cpp
@@ -9,9 +9,7 @@
 #include <minerfund.h>
 
 bool MinerFundPolicy::operator()(BlockPolicyValidationState &state) {
-    if (!m_blockIndex.pprev ||
-        !IsWellingtonEnabled(m_consensusParams, m_blockIndex.pprev)) {
-        // Do not apply the miner fund policy before Wellington activates.
+    if (!m_blockIndex.pprev) {
         return true;
     }
 
diff --git a/src/policy/block/stakingrewards.cpp b/src/policy/block/stakingrewards.cpp
index 952003a60..ca2e4f895 100644
--- a/src/policy/block/stakingrewards.cpp
+++ b/src/policy/block/stakingrewards.cpp
@@ -21,8 +21,7 @@
 static constexpr int STAKING_REWARD_RATIO = 10;
 
 bool StakingRewardsPolicy::operator()(BlockPolicyValidationState &state) {
-    if (!m_blockIndex.pprev ||
-        !IsCowperthwaiteEnabled(m_consensusParams, m_blockIndex.pprev)) {
+    if (!m_blockIndex.pprev) {
         return true;
     }
 
@@ -76,7 +75,7 @@ Amount GetStakingRewardsAmount(const Amount &coinbaseValue) {
 
 bool IsStakingRewardsActivated(const Consensus::Params &params,
                                const CBlockIndex *pprev) {
-    return IsCowperthwaiteEnabled(params, pprev) &&
+    return pprev != nullptr &&
            gArgs.GetBoolArg("-avalanchestakingrewards",
                             params.enableStakingRewards);
 }
diff --git a/src/pow/aserti32d.cpp b/src/pow/aserti32d.cpp
index eeaa9c797..b760ca729 100644
--- a/src/pow/aserti32d.cpp
+++ b/src/pow/aserti32d.cpp
@@ -11,7 +11,7 @@ uint32_t GetNextASERTWorkRequired(const CBlockIndex *pindexPrev,
                                   const Consensus::Params &params) noexcept {
     return GetNextASERTWorkRequired(
         pindexPrev, pblock, params,
-        pindexPrev->GetAncestor(params.axionHeight));
+        pindexPrev->GetAncestor(1));
 }
 
 /**
diff --git a/src/pow/pow.cpp b/src/pow/pow.cpp
index aedd5bd02..d365cff2e 100644
--- a/src/pow/pow.cpp
+++ b/src/pow/pow.cpp
@@ -31,15 +31,18 @@ uint32_t GetNextWorkRequired(const CBlockIndex *pindexPrev,
         return pindexPrev->nBits;
     }
 
-    if (IsAxionEnabled(params, pindexPrev)) {
-        return GetNextASERTWorkRequired(pindexPrev, pblock, params);
+    // ASERT requires an anchor block which has one parent.
+    // The Genesis block has no parent. Therefore, the block after Genesis
+    // will be our anchor block. We assign it minimum difficulty.
+    if (!pindexPrev->pprev) {
+        const uint32_t nProofOfWorkLimit =
+            UintToArith256(params.powLimit).GetCompact();
+        return nProofOfWorkLimit;
     }
 
-    if (IsDAAEnabled(params, pindexPrev)) {
-        return GetNextDAAWorkRequired(pindexPrev, pblock, params);
-    }
-
-    return GetNextEDAWorkRequired(pindexPrev, pblock, params);
+    // All blocks after genesis+1 adjust difficulty based on the genesis
+    // block timestamp as anchor.
+    return GetNextASERTWorkRequired(pindexPrev, pblock, params);
 }
 
 // Check that on difficulty adjustments, the new difficulty does not increase
@@ -60,28 +63,14 @@ bool PermittedDifficultyTransition(const Consensus::Params &params,
         return true;
     }
 
-    // Prior to the UAHF, the difficulty could change only every 2016 blocks,
-    // so we can bail out early if we observe a difficulty change at an
-    // unexpected block height.
-    if (!IsUAHFenabled(params, height - 1) &&
-        height % params.DifficultyAdjustmentInterval() != 0) {
-        return false;
-    }
-
     arith_uint256 observed_new_target;
-    // Check [0, powLimit] range for all DAA algorithms.
+    // Check [0, powLimit] range.
     if (!NBitsToTarget(params, new_nbits, observed_new_target)) {
         return false;
     }
 
-    // The newer difficulty adjustement algorithms (CW-144 and Aserti32d) do
-    // not have other hard rules for permitted difficulty transitions.
-    if (IsDAAEnabled(params, height - 1)) {
-        return true;
-    }
-
-    return PermittedEDADifficultyTransition(params, old_nbits,
-                                            observed_new_target);
+    // ASERT allows any difficulty transition.
+    return true;
 }
 
 bool CheckProofOfWork(const BlockHash &hash, uint32_t nBits,
diff --git a/src/pow/test/aserti32d_tests.cpp b/src/pow/test/aserti32d_tests.cpp
index 373b43e1b..e44c33a03 100644
--- a/src/pow/test/aserti32d_tests.cpp
+++ b/src/pow/test/aserti32d_tests.cpp
@@ -534,175 +534,4 @@ BOOST_AUTO_TEST_CASE(calculate_asert_test) {
     }
 }
 
-class ChainParamsWithCustomActivation : public CChainParams {
-public:
-    ChainParamsWithCustomActivation(const CChainParams &chainParams,
-                                    int daaHeight, int axionHeight)
-        : CChainParams(chainParams) {
-        BOOST_REQUIRE_GT(axionHeight, daaHeight);
-        consensus.daaHeight = daaHeight;
-        consensus.axionHeight = axionHeight;
-    }
-};
-
-/**
- * Test transition of cw144 to ASERT algorithm, which involves the selection
- * of an anchor block.
- */
-BOOST_AUTO_TEST_CASE(asert_activation_anchor_test) {
-    // Make a custom chain params based on mainnet, activating the cw144 DAA
-    // at a lower height than usual, so we don't need to waste time making a
-    // 504000-long chain.
-    const auto mainChainParams =
-        CreateChainParams(*m_node.args, ChainType::MAIN);
-    const int asertActivationHeight = 4000;
-    const ChainParamsWithCustomActivation chainParams(*mainChainParams, 2016,
-                                                      asertActivationHeight);
-    const Consensus::Params &params = chainParams.GetConsensus();
-
-    CBlockHeader blkHeaderDummy;
-
-    // an arbitrary compact target for our chain (based on BCH chain ~ Aug 10
-    // 2020).
-    uint32_t initialBits = 0x1802a842;
-
-    // Block store for anonymous blocks; needs to be big enough to fit all
-    // generated blocks in this test.
-    std::vector<CBlockIndex> blocks(10000);
-    int bidx = 1;
-
-    // Genesis block.
-    blocks[0].nHeight = 0;
-    blocks[0].nTime = 1269211443;
-    blocks[0].nBits = initialBits;
-    blocks[0].nChainWork = GetBlockProof(blocks[0]);
-
-    // Pile up a random number of blocks to establish some history of random
-    // height. cw144 DAA requires us to have height at least 2016, dunno why
-    // that much. Keep going up to 145 blocks prior to ASERT activation.
-    for (int i = 1; i < asertActivationHeight - 145; i++) {
-        BOOST_REQUIRE(bidx < int(blocks.size()));
-        blocks[bidx] = GetBlockIndex(&blocks[bidx - 1], 600, initialBits);
-        bidx++;
-    }
-    // Then put down 145 more blocks with 500 second solvetime each, such that
-    // the final block is the one prior to activation.
-    for (int i = 0; i < 145; i++) {
-        BOOST_REQUIRE(bidx < int(blocks.size()));
-        blocks[bidx] = GetBlockIndex(&blocks[bidx - 1], 500, initialBits);
-        bidx++;
-    }
-    CBlockIndex *pindexPreActivation = &blocks[bidx - 1];
-    BOOST_CHECK_EQUAL(pindexPreActivation->nHeight, asertActivationHeight - 1);
-    BOOST_CHECK(IsDAAEnabled(params, pindexPreActivation));
-
-    // If we consult DAA, then it uses cw144 which returns a significantly lower
-    // target because we have been mining too fast by a ratio 600/500 for a
-    // whole day.
-    BOOST_CHECK(!IsAxionEnabled(params, pindexPreActivation));
-    BOOST_CHECK_EQUAL(
-        GetNextWorkRequired(pindexPreActivation, &blkHeaderDummy, chainParams),
-        0x180236e1);
-
-    /**
-     * Now we'll try adding on blocks to activate ASERT. The activation block
-     * is going to be our anchor block. We will make several distinct anchor
-     * blocks.
-     */
-
-    // Create an activating block with expected solvetime, taking the cw144
-    // difficulty we just saw. Since solvetime is expected the next target is
-    // unchanged.
-    CBlockIndex indexActivation0 =
-        GetBlockIndex(pindexPreActivation, 600, 0x180236e1);
-    BOOST_CHECK(IsAxionEnabled(params, &indexActivation0));
-    BOOST_CHECK_EQUAL(
-        GetNextWorkRequired(&indexActivation0, &blkHeaderDummy, chainParams),
-        0x180236e1);
-    BOOST_CHECK_EQUAL(
-        GetNextWorkRequired(&indexActivation0, &blkHeaderDummy, chainParams),
-        0x180236e1);
-
-    // Now we'll generate some more activations/anchors, using unique targets
-    // for each one (if the algo gets confused between different anchors, we
-    // will know).
-
-    // Create an activating block with 0 solvetime, which will drop target by
-    // ~415/416.
-    CBlockIndex indexActivation1 =
-        GetBlockIndex(pindexPreActivation, 0, 0x18023456);
-    BOOST_CHECK(IsAxionEnabled(params, &indexActivation1));
-    // cache will be stale here, and we should get the right result regardless:
-    BOOST_CHECK_EQUAL(
-        GetNextWorkRequired(&indexActivation1, &blkHeaderDummy, chainParams),
-        0x180232fd);
-    BOOST_CHECK_EQUAL(
-        GetNextWorkRequired(&indexActivation1, &blkHeaderDummy, chainParams),
-        0x180232fd);
-    BOOST_CHECK_EQUAL(
-        GetNextWorkRequired(&indexActivation1, &blkHeaderDummy, chainParams),
-        0x180232fd);
-
-    // Try activation with expected solvetime, which will keep target the same.
-    uint32_t anchorBits2 = 0x180210fe;
-    CBlockIndex indexActivation2 =
-        GetBlockIndex(pindexPreActivation, 600, anchorBits2);
-    BOOST_CHECK(IsAxionEnabled(params, &indexActivation2));
-    BOOST_CHECK_EQUAL(
-        GetNextWorkRequired(&indexActivation2, &blkHeaderDummy, chainParams),
-        anchorBits2);
-
-    // Try a three-month solvetime which will cause us to hit powLimit.
-    uint32_t anchorBits3 = 0x18034567;
-    CBlockIndex indexActivation3 =
-        GetBlockIndex(pindexPreActivation, 86400 * 90, anchorBits3);
-    BOOST_CHECK(IsAxionEnabled(params, &indexActivation2));
-    BOOST_CHECK_EQUAL(
-        GetNextWorkRequired(&indexActivation3, &blkHeaderDummy, chainParams),
-        0x1d00ffff);
-    // If the next block jumps back in time, we get back our original difficulty
-    // level.
-    CBlockIndex indexActivation3_return =
-        GetBlockIndex(&indexActivation3, -86400 * 90 + 2 * 600, anchorBits3);
-    BOOST_CHECK_EQUAL(GetNextWorkRequired(&indexActivation3_return,
-                                          &blkHeaderDummy, chainParams),
-                      anchorBits3);
-    // Retry for cache
-    BOOST_CHECK_EQUAL(GetNextWorkRequired(&indexActivation3_return,
-                                          &blkHeaderDummy, chainParams),
-                      anchorBits3);
-
-    // Make an activation with MTP == activation exactly. This is a backwards
-    // timestamp jump so the resulting target is 1.2% lower.
-    CBlockIndex indexActivation4 =
-        GetBlockIndex(pindexPreActivation, 0, 0x18011111);
-    indexActivation4.nTime = pindexPreActivation->GetMedianTimePast();
-    BOOST_CHECK(IsAxionEnabled(params, &indexActivation4));
-    BOOST_CHECK_EQUAL(
-        GetNextWorkRequired(&indexActivation4, &blkHeaderDummy, chainParams),
-        0x18010db3);
-
-    // Finally create a random chain on top of our second activation, using
-    // ASERT targets all the way. Erase cache so that this will do a fresh
-    // search for anchor at every step (fortauntely this is not too slow, due to
-    // the skiplist traversal)
-    CBlockIndex *pindexChain2 = &indexActivation2;
-    for (int i = 1; i < 1000; i++) {
-        BOOST_REQUIRE(bidx < int(blocks.size()));
-        uint32_t nextBits =
-            GetNextWorkRequired(pindexChain2, &blkHeaderDummy, chainParams);
-        blocks[bidx] =
-            GetBlockIndex(pindexChain2, m_rng.randrange(1200), nextBits);
-        pindexChain2 = &blocks[bidx++];
-    }
-    // Scan back down to make sure all targets are same when we keep cached
-    // anchor.
-    for (CBlockIndex *pindex = pindexChain2; pindex != &indexActivation2;
-         pindex = pindex->pprev) {
-        uint32_t nextBits =
-            GetNextWorkRequired(pindex->pprev, &blkHeaderDummy, chainParams);
-        BOOST_CHECK_EQUAL(nextBits, pindex->nBits);
-    }
-}
-
 BOOST_AUTO_TEST_SUITE_END()
diff --git a/src/rpc/mining.cpp b/src/rpc/mining.cpp
index d69171039..b0afc23ad 100644
--- a/src/rpc/mining.cpp
+++ b/src/rpc/mining.cpp
@@ -1129,13 +1129,10 @@ static RPCHelpMan getblocktemplate() {
             // Compute the miner fund parameters
             const auto minerFundWhitelist =
                 GetMinerFundWhitelist(consensusParams);
-            int64_t minerFundMinValue = 0;
-            if (IsAxionEnabled(consensusParams, pindexPrev)) {
-                minerFundMinValue =
-                    int64_t(GetMinerFundAmount(consensusParams, coinbasevalue,
-                                               pindexPrev) /
-                            SATOSHI);
-            }
+            int64_t minerFundMinValue =
+                int64_t(GetMinerFundAmount(consensusParams, coinbasevalue,
+                                           pindexPrev) /
+                        SATOSHI);
 
             // Compute the staking reward parameters
             std::vector<CScript> stakingRewardsPayoutScripts;
diff --git a/src/test/CMakeLists.txt b/src/test/CMakeLists.txt
index 9c5a6db6a..3afac158c 100644
--- a/src/test/CMakeLists.txt
+++ b/src/test/CMakeLists.txt
@@ -311,8 +311,6 @@ function(add_boost_test_runners_with_upgrade_activated SUITE EXECUTABLE)
 			--
 			"-printtoconsole=1"
 			"-testsuitename=Bitcoin ABC unit tests with next upgrade activated"
-			# May. 16th, 2025 at 08:00:00
-			-mengeractivationtime=1747382400
 		)
 	endforeach()
 endfunction()
diff --git a/src/test/activation_tests.cpp b/src/test/activation_tests.cpp
index 0d352dd9f..0068addea 100644
--- a/src/test/activation_tests.cpp
+++ b/src/test/activation_tests.cpp
@@ -2,59 +2,14 @@
 // Distributed under the MIT software license, see the accompanying
 // file COPYING or http://www.opensource.org/licenses/mit-license.php.
 
-#include <chain.h>
-#include <chainparams.h>
-#include <common/system.h>
-#include <config.h>
-#include <consensus/activation.h>
-#include <script/script.h>
-#include <txmempool.h>
-#include <util/chaintype.h>
-#include <util/check.h>
-#include <validation.h>
-
-#include <test/util/blockindex.h>
-#include <test/util/mining.h>
 #include <test/util/setup_common.h>
 
 #include <boost/test/unit_test.hpp>
 
 BOOST_FIXTURE_TEST_SUITE(activation_tests, BasicTestingSetup)
 
-using ActivationFun = bool (*)(const Consensus::Params &, const CBlockIndex *);
-
-static void testPastActivation(ActivationFun func,
-                               const Consensus::Params &params,
-                               int activationHeight) {
-    BOOST_CHECK(!func(params, nullptr));
-
-    std::array<CBlockIndex, 4> blocks;
-    blocks[0].nHeight = activationHeight - 2;
-    for (size_t i = 1; i < blocks.size(); ++i) {
-        blocks[i].pprev = &blocks[i - 1];
-        blocks[i].nHeight = blocks[i - 1].nHeight + 1;
-    }
-
-    BOOST_CHECK(!func(params, &blocks[0]));
-    BOOST_CHECK(!func(params, &blocks[1]));
-    BOOST_CHECK(func(params, &blocks[2]));
-    BOOST_CHECK(func(params, &blocks[3]));
-}
-
-BOOST_AUTO_TEST_CASE(test_previous_activations_by_height) {
-    const auto params = CreateChainParams(*m_node.args, ChainType::MAIN);
-    const auto consensus = params->GetConsensus();
-
-    // Static cast to select the correct overload
-    testPastActivation(static_cast<ActivationFun>(IsMagneticAnomalyEnabled),
-                       consensus, consensus.magneticAnomalyHeight);
-    testPastActivation(IsGravitonEnabled, consensus, consensus.gravitonHeight);
-    testPastActivation(IsPhononEnabled, consensus, consensus.phononHeight);
-    testPastActivation(IsAxionEnabled, consensus, consensus.axionHeight);
-    testPastActivation(static_cast<ActivationFun>(IsWellingtonEnabled),
-                       consensus, consensus.wellingtonHeight);
-    testPastActivation(static_cast<ActivationFun>(IsCowperthwaiteEnabled),
-                       consensus, consensus.cowperthwaiteHeight);
+BOOST_AUTO_TEST_CASE(activationtest) {
+    /* Place future activation logic tests here. */
 }
 
 BOOST_AUTO_TEST_SUITE_END()
diff --git a/src/test/mempool_tests.cpp b/src/test/mempool_tests.cpp
index 06058654f..e741d76bc 100644
--- a/src/test/mempool_tests.cpp
+++ b/src/test/mempool_tests.cpp
@@ -595,7 +595,9 @@ BOOST_AUTO_TEST_CASE(remove_for_finalized_block) {
     Package txs;
     txs.reserve(100);
     for (size_t i = 0; i < 100; i++) {
-        CTransactionRef tx = make_tx({int64_t(i + 1) * COIN});
+        CMutableTransaction mtx(*make_tx({int64_t(i + 1) * COIN}));
+        mtx.vout[0].scriptPubKey << std::vector<uint8_t>(MIN_TX_SIZE, 0);
+        CTransactionRef tx = MakeTransactionRef(std::move(mtx));
         const TxId &txid = tx->GetId();
         auto mempoolEntry = entry.Fee(1000 * SATOSHI).FromTx(tx);
 
@@ -679,7 +681,7 @@ BOOST_AUTO_TEST_CASE(setAvalancheFinalized_ContextualCheckTransactionFailures) {
 
     // Fake chain tip
     CBlockIndex active_chain_tip;
-    active_chain_tip.nHeight = params.wellingtonHeight;
+    active_chain_tip.nHeight = 100;
 
     std::array<CBlockIndex, 12> blocks;
     for (size_t i = 0; i < blocks.size(); ++i) {
diff --git a/src/test/minerfund_tests.cpp b/src/test/minerfund_tests.cpp
index 94bbf49d2..5744a01fb 100644
--- a/src/test/minerfund_tests.cpp
+++ b/src/test/minerfund_tests.cpp
@@ -32,34 +32,19 @@ BOOST_AUTO_TEST_CASE(minerfund_whitelist) {
     const CChainParams &chainparams = Params();
     const Consensus::Params &consensusParams = chainparams.GetConsensus();
 
-    std::array<CBlockIndex, 3> blocks;
-
-    blocks[0].nHeight = 1;
-    blocks[1].nHeight = consensusParams.wellingtonHeight;
-    blocks[2].nHeight = consensusParams.wellingtonHeight + 1;
-
-    for (size_t i = 1; i < blocks.size(); ++i) {
-        blocks[i].pprev = &blocks[i - 1];
-    }
+    CBlockIndex block;
+    block.nHeight = 1;
 
     const std::unordered_set<CTxDestination, TxDestinationHasher>
         expectedMinerFund = {DecodeDestination(
             "ecash:prfhcnyqnl5cgrnmlfmms675w93ld7mvvqd0y8lz07", chainparams)};
-    CheckWhitelist(consensusParams, &blocks[0], expectedMinerFund);
-
-    // Test address does not change around Wellington activation
-    BOOST_CHECK(!IsWellingtonEnabled(consensusParams, blocks[1].pprev));
-    CheckWhitelist(consensusParams, &blocks[1], expectedMinerFund);
-
-    BOOST_CHECK(IsWellingtonEnabled(consensusParams, blocks[2].pprev));
-    CheckWhitelist(consensusParams, &blocks[2], expectedMinerFund);
+    CheckWhitelist(consensusParams, &block, expectedMinerFund);
 }
 
 BOOST_AUTO_TEST_CASE(minerfund_amount) {
     const CChainParams &chainparams = Params();
     const Consensus::Params &consensusParams = chainparams.GetConsensus();
 
-    const auto activationHeight = consensusParams.cowperthwaiteHeight;
     CBlockIndex block;
 
     auto checkMinerFundRatio = [&](int64_t expectedPercent) {
@@ -68,18 +53,9 @@ BOOST_AUTO_TEST_CASE(minerfund_amount) {
             expectedPercent * COIN);
     };
 
-    // Before Cowperthwaite activation, the miner fund ratio is 8%
-    block.nHeight = activationHeight - 1000;
-    checkMinerFundRatio(8);
-    block.nHeight = activationHeight - 1;
-    checkMinerFundRatio(8);
-
-    // After Cowperthwaite activation, the miner fund ratio is 32%
-    block.nHeight = activationHeight;
-    checkMinerFundRatio(32);
-    block.nHeight = activationHeight + 1;
+    block.nHeight = 1;
     checkMinerFundRatio(32);
-    block.nHeight = activationHeight + 1000;
+    block.nHeight = 1000;
     checkMinerFundRatio(32);
 }
 
diff --git a/src/test/policy_block_tests.cpp b/src/test/policy_block_tests.cpp
index 548f78695..a6bbc3850 100644
--- a/src/test/policy_block_tests.cpp
+++ b/src/test/policy_block_tests.cpp
@@ -52,9 +52,8 @@ BOOST_AUTO_TEST_CASE(policy_minerfund) {
         blockhashes[i] = BlockHash(uint256(i));
         blocks[i].phashBlock = &blockhashes[i];
         blocks[i].pprev = &blocks[i - 1];
-        blocks[i].nHeight = consensusParams.wellingtonHeight - 5 + i;
+        blocks[i].nHeight = i;
     }
-    CBlockIndex &firstBlockIndexRef = blocks[1];
     CBlockIndex &lastBlockIndexRef = blocks.back();
 
     const Amount blockReward =
@@ -72,14 +71,6 @@ BOOST_AUTO_TEST_CASE(policy_minerfund) {
         }
     };
 
-    // Before wellington activation, the block policy is not enforced
-    BOOST_CHECK(
-        !IsWellingtonEnabled(consensusParams, firstBlockIndexRef.pprev));
-    checkMinerFundPolicy(BlockWithoutMinerFund(blockReward), firstBlockIndexRef,
-                         true);
-
-    // After wellington activation, the block policy is enforced
-    BOOST_CHECK(IsWellingtonEnabled(consensusParams, lastBlockIndexRef.pprev));
     checkMinerFundPolicy(BlockWithoutMinerFund(blockReward), lastBlockIndexRef,
                          false);
 
@@ -101,51 +92,10 @@ BOOST_AUTO_TEST_CASE(policy_minerfund) {
         checkMinerFundPolicy(BlockWithMinerFund(chainparams, amount),
                              lastBlockIndexRef, true);
     }
-
-    // Blocks with sufficient miner fund before Cowperthwaite activation...
     for (const Amount &amount : minerFundsSufficient) {
         checkMinerFundPolicy(BlockWithMinerFund(chainparams, amount),
                              lastBlockIndexRef, true);
     }
-
-    // ... but not after Cowperthwaite activation
-    CBlockIndex cowperthwaiteBlockIndex;
-    BlockHash cowperthwaiteBlockHash = BlockHash(uint256(13));
-    cowperthwaiteBlockIndex.phashBlock = &cowperthwaiteBlockHash;
-    cowperthwaiteBlockIndex.pprev = &lastBlockIndexRef;
-    lastBlockIndexRef.nHeight = consensusParams.cowperthwaiteHeight;
-    cowperthwaiteBlockIndex.nHeight = consensusParams.cowperthwaiteHeight + 1;
-
-    BOOST_CHECK(
-        IsCowperthwaiteEnabled(consensusParams, cowperthwaiteBlockIndex.pprev));
-
-    for (const Amount &amount : minerFundsSufficient) {
-        checkMinerFundPolicy(BlockWithMinerFund(chainparams, amount),
-                             cowperthwaiteBlockIndex, false);
-    }
-
-    // Update the miner fund values
-    minerFund = GetMinerFundAmount(consensusParams, blockReward,
-                                   cowperthwaiteBlockIndex.pprev);
-    minerFundsSufficient = {minerFund, minerFund + 1 * SATOSHI};
-
-    // Blocks with sufficient miner fund after Cowperthwaite activation
-    for (const Amount &amount : minerFundsSufficient) {
-        checkMinerFundPolicy(BlockWithMinerFund(chainparams, amount),
-                             cowperthwaiteBlockIndex, true);
-    }
-
-    // Sanity check the always rejected/accepted blocks are unchanged
-    // Blocks with not enough miner fund are always rejected
-    for (const Amount &amount : minerFundsTooSmall) {
-        checkMinerFundPolicy(BlockWithMinerFund(chainparams, amount),
-                             cowperthwaiteBlockIndex, false);
-    }
-    // Blocks with enough miner fund are always accepted
-    for (const Amount &amount : minerFundsAlwaysSufficient) {
-        checkMinerFundPolicy(BlockWithMinerFund(chainparams, amount),
-                             cowperthwaiteBlockIndex, true);
-    }
 }
 
 BOOST_AUTO_TEST_SUITE_END()
diff --git a/src/test/transaction_tests.cpp b/src/test/transaction_tests.cpp
index edbcc9530..00433ad43 100644
--- a/src/test/transaction_tests.cpp
+++ b/src/test/transaction_tests.cpp
@@ -890,17 +890,12 @@ BOOST_AUTO_TEST_CASE(test_IsStandard) {
 
 BOOST_FIXTURE_TEST_CASE(txsize_activation_test, ChainTestingSetup) {
     const Consensus::Params &params = m_node.chainman->GetConsensus();
-    const int32_t magneticAnomalyActivationHeight =
-        params.magneticAnomalyHeight;
 
     // A minimaly sized transction.
     CTransaction minTx;
     TxValidationState state;
 
-    BOOST_CHECK(ContextualCheckTransaction(
-        params, minTx, state, magneticAnomalyActivationHeight - 1, 5678));
-    BOOST_CHECK(!ContextualCheckTransaction(
-        params, minTx, state, magneticAnomalyActivationHeight, 5678));
+    BOOST_CHECK(!ContextualCheckTransaction(params, minTx, state, 0, 5678));
     BOOST_CHECK_EQUAL(state.GetRejectReason(), "bad-txns-undersize");
 }
 
diff --git a/src/validation.cpp b/src/validation.cpp
index 2a45e7375..e1fda885d 100644
--- a/src/validation.cpp
+++ b/src/validation.cpp
@@ -223,7 +223,7 @@ IsReplayProtectionEnabled(const Consensus::Params &params,
     }
 
     return pindexPrev->GetMedianTimePast() >=
-           activation_time.value_or(params.sandersActivationTime);
+           activation_time.value_or(0xffffffff);
 }
 
 /**
@@ -2028,60 +2028,14 @@ static uint32_t GetNextBlockScriptFlags(const CBlockIndex *pindex,
                                         const ChainstateManager &chainman) {
     const Consensus::Params &consensusparams = chainman.GetConsensus();
 
-    uint32_t flags = SCRIPT_VERIFY_NONE;
-
-    // Enforce P2SH (BIP16)
-    if (DeploymentActiveAfter(pindex, chainman, Consensus::DEPLOYMENT_P2SH)) {
-        flags |= SCRIPT_VERIFY_P2SH;
-    }
-
-    // Enforce the DERSIG (BIP66) rule.
-    if (DeploymentActiveAfter(pindex, chainman, Consensus::DEPLOYMENT_DERSIG)) {
-        flags |= SCRIPT_VERIFY_DERSIG;
-    }
-
-    // Start enforcing CHECKLOCKTIMEVERIFY (BIP65) rule.
-    if (DeploymentActiveAfter(pindex, chainman, Consensus::DEPLOYMENT_CLTV)) {
-        flags |= SCRIPT_VERIFY_CHECKLOCKTIMEVERIFY;
-    }
-
-    // Start enforcing CSV (BIP68, BIP112 and BIP113) rule.
-    if (DeploymentActiveAfter(pindex, chainman, Consensus::DEPLOYMENT_CSV)) {
-        flags |= SCRIPT_VERIFY_CHECKSEQUENCEVERIFY;
-    }
-
-    // If the UAHF is enabled, we start accepting replay protected txns
-    if (IsUAHFenabled(consensusparams, pindex)) {
-        flags |= SCRIPT_VERIFY_STRICTENC;
-        flags |= SCRIPT_ENABLE_SIGHASH_FORKID;
-    }
-
-    // If the DAA HF is enabled, we start rejecting transaction that use a high
-    // s in their signature. We also make sure that signature that are supposed
-    // to fail (for instance in multisig or other forms of smart contracts) are
-    // null.
-    if (IsDAAEnabled(consensusparams, pindex)) {
-        flags |= SCRIPT_VERIFY_LOW_S;
-        flags |= SCRIPT_VERIFY_NULLFAIL;
-    }
-
-    // When the magnetic anomaly fork is enabled, we start accepting
-    // transactions using the OP_CHECKDATASIG opcode and it's verify
-    // alternative. We also start enforcing push only signatures and
-    // clean stack.
-    if (IsMagneticAnomalyEnabled(consensusparams, pindex)) {
-        flags |= SCRIPT_VERIFY_SIGPUSHONLY;
-        flags |= SCRIPT_VERIFY_CLEANSTACK;
-    }
-
-    if (IsGravitonEnabled(consensusparams, pindex)) {
-        flags |= SCRIPT_ENABLE_SCHNORR_MULTISIG;
-        flags |= SCRIPT_VERIFY_MINIMALDATA;
-    }
-
-    if (IsPhononEnabled(consensusparams, pindex)) {
-        flags |= SCRIPT_ENFORCE_SIGCHECKS;
-    }
+    uint32_t flags = SCRIPT_VERIFY_P2SH | SCRIPT_VERIFY_DERSIG |
+                     SCRIPT_VERIFY_CHECKLOCKTIMEVERIFY |
+                     SCRIPT_VERIFY_CHECKSEQUENCEVERIFY |
+                     SCRIPT_VERIFY_STRICTENC | SCRIPT_ENABLE_SIGHASH_FORKID |
+                     SCRIPT_VERIFY_LOW_S | SCRIPT_VERIFY_NULLFAIL |
+                     SCRIPT_VERIFY_SIGPUSHONLY | SCRIPT_VERIFY_CLEANSTACK |
+                     SCRIPT_ENABLE_SCHNORR_MULTISIG |
+                     SCRIPT_VERIFY_MINIMALDATA | SCRIPT_ENFORCE_SIGCHECKS;
 
     // We make sure this node will have replay protection during the next hard
     // fork.
@@ -4496,22 +4450,6 @@ static bool ContextualCheckBlockHeader(
                              "block timestamp too far in the future");
     }
 
-    // Reject blocks with outdated version
-    if ((block.nVersion < 2 &&
-         DeploymentActiveAfter(pindexPrev, chainman,
-                               Consensus::DEPLOYMENT_HEIGHTINCB)) ||
-        (block.nVersion < 3 &&
-         DeploymentActiveAfter(pindexPrev, chainman,
-                               Consensus::DEPLOYMENT_DERSIG)) ||
-        (block.nVersion < 4 &&
-         DeploymentActiveAfter(pindexPrev, chainman,
-                               Consensus::DEPLOYMENT_CLTV))) {
-        return state.Invalid(
-            BlockValidationResult::BLOCK_INVALID_HEADER,
-            strprintf("bad-version(0x%08x)", block.nVersion),
-            strprintf("rejected nVersion=0x%08x block", block.nVersion));
-    }
-
     return true;
 }
 
@@ -4528,24 +4466,13 @@ static bool ContextualCheckBlock(const CBlock &block,
                                  const CBlockIndex *pindexPrev) {
     const int nHeight = pindexPrev == nullptr ? 0 : pindexPrev->nHeight + 1;
 
-    // Enforce BIP113 (Median Time Past).
-    bool enforce_locktime_median_time_past{false};
-    if (DeploymentActiveAfter(pindexPrev, chainman,
-                              Consensus::DEPLOYMENT_CSV)) {
-        assert(pindexPrev != nullptr);
-        enforce_locktime_median_time_past = true;
-    }
-
+    // Always enforce BIP113 (Median Time Past) from genesis.
     const int64_t nMedianTimePast =
         pindexPrev == nullptr ? 0 : pindexPrev->GetMedianTimePast();
 
-    const int64_t nLockTimeCutoff{enforce_locktime_median_time_past
-                                      ? nMedianTimePast
-                                      : block.GetBlockTime()};
+    const int64_t nLockTimeCutoff = nMedianTimePast;
 
     const Consensus::Params params = chainman.GetConsensus();
-    const bool fIsMagneticAnomalyEnabled =
-        IsMagneticAnomalyEnabled(params, pindexPrev);
 
     // Check transactions:
     // - canonical ordering
@@ -4554,25 +4481,23 @@ static bool ContextualCheckBlock(const CBlock &block,
     const CTransaction *prevTx = nullptr;
     for (const auto &ptx : block.vtx) {
         const CTransaction &tx = *ptx;
-        if (fIsMagneticAnomalyEnabled) {
-            if (prevTx && (tx.GetId() <= prevTx->GetId())) {
-                if (tx.GetId() == prevTx->GetId()) {
-                    return state.Invalid(BlockValidationResult::BLOCK_CONSENSUS,
-                                         "tx-duplicate",
-                                         strprintf("Duplicated transaction %s",
-                                                   tx.GetId().ToString()));
-                }
-
-                return state.Invalid(
-                    BlockValidationResult::BLOCK_CONSENSUS, "tx-ordering",
-                    strprintf("Transaction order is invalid (%s < %s)",
-                              tx.GetId().ToString(),
-                              prevTx->GetId().ToString()));
+        if (prevTx && (tx.GetId() <= prevTx->GetId())) {
+            if (tx.GetId() == prevTx->GetId()) {
+                return state.Invalid(BlockValidationResult::BLOCK_CONSENSUS,
+                                     "tx-duplicate",
+                                     strprintf("Duplicated transaction %s",
+                                               tx.GetId().ToString()));
             }
 
-            if (prevTx || !tx.IsCoinBase()) {
-                prevTx = &tx;
-            }
+            return state.Invalid(
+                BlockValidationResult::BLOCK_CONSENSUS, "tx-ordering",
+                strprintf("Transaction order is invalid (%s < %s)",
+                          tx.GetId().ToString(),
+                          prevTx->GetId().ToString()));
+        }
+
+        if (prevTx || !tx.IsCoinBase()) {
+            prevTx = &tx;
         }
 
         TxValidationState tx_state;
@@ -4585,8 +4510,7 @@ static bool ContextualCheckBlock(const CBlock &block,
     }
 
     // Enforce rule that the coinbase starts with serialized block height
-    if (DeploymentActiveAfter(pindexPrev, chainman,
-                              Consensus::DEPLOYMENT_HEIGHTINCB)) {
+    if (nHeight >= 1) {
         CScript expect = CScript() << nHeight;
         if (block.vtx[0]->vin[0].scriptSig.size() < expect.size() ||
             !std::equal(expect.begin(), expect.end(),
-- 
2.50.1 (Apple Git-155)

